### DISCUSSION TITLE
Finland: grid charge (Elenia) + electricity as spot + margin (Sahkolaitos)

### DISCUSSION THREAD

**Discussion by jyrkih** [JaccoR/hass-entso-e#39] [Price Modifyer Templates]
```
{% set s = {
    "vat": 1.24
}
%}
{% set time = { "hour": (now().strftime('%H') | int) } %}
{% if (time.hour >= 7 ) and (time.hour < 22 ) %}
  {% set grid_price = 0.0680 %}
{% else %}
  {% set grid_price = 0.0524 %}
{% endif %}

{% set local_tariff = { "T2022": 0.0037944, "T2023A": 0.0068 } %}
{% set time = { "month": (now().strftime('%m') | int), "day": (now().strftime('%d') | int), "year": (now().strftime('%Y') | int ) } %}
{% if (time.year <= 2022) and (time.month < 11)%}
  {% set local_margin = local_tariff.T2022 %}
{% else %}
  {% set local_margin = local_tariff.T2023A %}
{% endif %}
{% if (time.year >= 2022) and (time.month > 11) and (time.year <= 2023) and (time.month < 5) %}
  {% set s.vat = 1.10 %}
{% endif %}

{{(current_price * s.vat) + grid_price + local_margin | float}}
```
grid_price with peak (7-22) and offpeak hours, local electricity company margin as local_margin
temporary VAT 10% during 12/22 - 04/23

**Comment by JaccoR** [comment]
In the new release, VAT can be added as an config option. This means the template does not need the VAT anymore.

**Comment by lazmo88** [comment]
@jyrkih this is great 👌 Few questions tho.

1. Are grid_prices vat exclusive or vat inclusive values? It seems that template doesn't calculate VAT on top, so I assuming they are VAT inclusive?

2. I noted local_margin has a conditional clause, most likely due to known price increase after the November 2022? I wondered if this could be easier to manage via dashboard input instead of hard coded template config 🤔

**Reply by jyrkih** [reply]
Hi, yes regarding VAT, prices in Finland include VAT, so it's easier to use that approach. So for this reason, I can't use input VAT either, but set that to 1 and add  VAT only to nordpool price.

There are known price changes, that template tries to tackle those. But I agree, there could be a better way.
